fast_api_als/utils/adf.py

```diff
@@ -39,7 +47,12 @@ def is_nan(x):
 
 def parse_xml(adf_xml):
     # use exception handling
-    obj = xmltodict.parse(adf_xml)
+    try:
+        obj = xmltodict.parse(adf_xml)
+        info(" successfully prase xml file")
+    except Exception as e:
+        error("Unable to parse xml file" + e)
+        raise Exception('Unable to parse adf xml')
```

Comment:
If you're raising the Exception, then make sure you're catching it on where you called this function.
Check the logic of the code that is calling this function, and decide whether you should throw the Exception or do something else.